modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java

```diff
@@ -552,23 +552,24 @@ protected QueryFilter createFieldValueQueryFilter(
 
 		BooleanQuery booleanQuery = new BooleanQueryImpl();
 
-		if (!isLegacyDDMIndexFieldsEnabled()) {
-			booleanQuery.addRequiredTerm(
-				StringBundler.concat(
-					DDM_FIELD_ARRAY, StringPool.PERIOD, DDM_FIELD_NAME),
-				ddmStructureFieldName);
+		if (isLegacyDDMIndexFieldsEnabled()) {
+			_addFieldValueRequiredTerm(
+				booleanQuery, ddmStructureFieldName, ddmStructureFieldValue);
 
-			ddmStructureFieldName = StringBundler.concat(
-				DDM_FIELD_ARRAY, StringPool.PERIOD,
-				getValueFieldName(indexType, locale));
+			return new QueryFilter(booleanQuery);
 		}
 
-		_addFieldValueRequiredTerm(
-			booleanQuery, ddmStructureFieldName, ddmStructureFieldValue);
+		booleanQuery.addRequiredTerm(
+			StringBundler.concat(
+				DDM_FIELD_ARRAY, StringPool.PERIOD, DDM_FIELD_NAME),
+			ddmStructureFieldName);
 
-		if (isLegacyDDMIndexFieldsEnabled()) {
-			return new QueryFilter(booleanQuery);
-		}
+		_addFieldValueRequiredTerm(
```

Comment:
Hi @jorgediaz-lr,
I think we can put this code:
```
boolean localizable = false;

if (ddmStructure.hasFieldByFieldReference(fieldReference)) {
	ddmStructureFieldValue = _ddm.getIndexedFieldValue(
		ddmStructureFieldValue,
		ddmStructure.getFieldPropertyByFieldReference(
			fieldReference, "type"));
	localizable = GetterUtil.getBoolean(
		ddmStructure.getFieldPropertyByFieldReference(
			fieldReference, "localizable"));
}

if (!localizable) {
	locale = null;
}
```
(line 536 to 551) right before this line (567) since the `locale` is only being used by `getValueFieldName`. 

What do you think?

Reply:
Hi @carolmariaabb 

We cannot move the complete code block, because **ddmStructureFieldValue** is updated with the result of `_ddm.getIndexedFieldValue` and it is used in line 557, before the line 567 you said.

We can move this part:
```
	localizable = GetterUtil.getBoolean(
		ddmStructure.getFieldPropertyByFieldReference(
			fieldReference, "localizable"));
}

if (!localizable) {
	locale = null;
}
```

But the first line **ddmStructureFieldValue = _ddm.getIndexedFieldValue(** cannot be moved.

Let me know if you want to do this change.

Reply:
Got i! Let's keep it the way it's then. Thanks!